engine/src/main/java/org/terasology/engine/physics/engine/PhysicsSystem.java

```diff
@@ -164,6 +195,125 @@ public void onItemImpact(ImpactEvent event, EntityRef entity) {
+        meshData.position.put(pos.set(bounds.maxX, bounds.minY, bounds.minZ));
+        meshData.position.put(pos.set(bounds.maxX, bounds.minY, bounds.maxZ));
+        meshData.position.put(pos.set(bounds.minX, bounds.minY, bounds.maxZ));
+
+        meshData.position.put(pos.set(bounds.minX, bounds.maxY, bounds.minZ));
+        meshData.position.put(pos.set(bounds.maxX, bounds.maxY, bounds.minZ));
+        meshData.position.put(pos.set(bounds.maxX, bounds.maxY, bounds.maxZ));
+        meshData.position.put(pos.set(bounds.minX, bounds.maxY, bounds.maxZ));
+
+        meshData.color0.put(Color.black);
+        meshData.color0.put(Color.black);
+        meshData.color0.put(Color.black);
+        meshData.color0.put(Color.black);
+
+        meshData.color0.put(Color.black);
+        meshData.color0.put(Color.black);
+        meshData.color0.put(Color.black);
+        meshData.color0.put(Color.black);
+
+        meshData.indices.putAll(new int[]{
+                // top loop
+                index, index + 1,
+                index + 1, index + 2,
+                index + 2, index + 3,
+                index + 3, index,
+
+                // connecting edges between top and bottom
+                index, index + 4,
+                index + 1, index + 5,
+                index + 2, index + 6,
+                index + 3, index + 7,
+
+                // bottom loop
+                index + 4, index + 5,
+                index + 5, index + 6,
+                index + 6, index + 7,
+                index + 7, index + 4,
+
+        });
+        return index + 8;
+    }
+
+
+
+    @Override
+    public void renderOverlay() {
+        if (config.getRendering().getDebug().isRenderingEntityColliders()) {
+            GL33.glDepthFunc(GL33.GL_ALWAYS);
+            Vector3f cameraPosition = worldRenderer.getActiveCamera().getPosition();
+
+            Vector3f worldPos = new Vector3f();
+            Vector3f worldPositionCameraSpace = new Vector3f();
+            worldPos.sub(cameraPosition, worldPositionCameraSpace);
+            Matrix4f matrixCameraSpace = new Matrix4f().translationRotateScale(worldPositionCameraSpace, new Quaternionf(), 1.0f);
+            Matrix4f modelViewMatrix = new Matrix4f(worldRenderer.getActiveCamera().getViewMatrix()).mul(matrixCameraSpace);
+            material.setMatrix4("projectionMatrix", worldRenderer.getActiveCamera().getProjectionMatrix());
+            material.setMatrix4("modelViewMatrix", modelViewMatrix, true);
+
+            int index = 0;
+            meshData.reallocate(0, 0);
+            meshData.indices.rewind();
+            meshData.position.rewind();
+            meshData.color0.rewind();
+
+            Vector3f worldPosition = new Vector3f();
+            Quaternionf worldRot = new Quaternionf();
+            Matrix4f transform = new Matrix4f();
+            AABBf bounds = new AABBf(0, 0, 0, 0, 0, 0);
+
+            for (EntityRef entity : entityManager.getEntitiesWith(RigidBodyComponent.class)) {
```

Comment:
Not all the entities have a `RigidBodyComponent` - should this rather be for `LocationComponent`?